src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -20,227 +129,2330 @@ const ActorInit En_Ossan_InitVars = {
+        this->unk_284 = 1.0f;
+        this->unk_2BC = 1.0f;
+        this->unk_264 = 49.0f;
+        this->unk_27C = 33.0f;
+        this->unk_24C = 0.0f;
+        this->unk_2C4 = 0.0f;
+        this->unk_2C8 = 0.0f;
+        this->unk_2D0 = 0.0f;
+        this->unk_29C = 274.0f;
+        this->unk_2B4 = 290.0f;
+
+        Actor_SetScale(&this->actor, D_80AC89B0[this->actor.params]);
+        func_80AC2DD8(this, globalCtx, sp24);
+
+        this->unk_1F4 = 0x14;
+        this->unk_1F2 = 0;
+        this->unk_1F8 = func_80AC6970;
+        this->actor.flags &= ~-0x1;
+        this->unk_1EE = this->unk_1F0 = 0;
+        func_80AC2CA0(this, func_80AC73B4);
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC7094.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC7380.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC73B4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/EnOssan_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC74F4.s")
-
+#endif
+
+void func_80AC7380(EnOssan* this, GlobalContext* globalCtx) {
+    gSegments[6] = PHYSICAL_TO_VIRTUAL(globalCtx->objectCtx.status[this->unk_1EA].segment);
+}
+
+void func_80AC73B4(EnOssan* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    Actor* thisx = &this->actor;
+
+    this->unk_1F8(this);
+    func_80AC3BCC(globalCtx, this);
+    func_80AC65B8(this);
+    func_80AC67C4(this);
+    func_80AC66F4(this);
+    Math_ApproxS(&this->unk_1EE, this->unk_1F0, 0x190);
+    if (player != NULL) {
+        D_80AC8E34[this->unk_1FC](this, globalCtx, player);
+    }
+    Actor_MoveForward(thisx);
+    func_8002E4B4(globalCtx, thisx, 26.0f, 10.0f, 0.0f, 5);
+    Actor_SetHeight(&this->actor, 90.0f);
+    Actor_SetScale(&this->actor, D_80AC89B0[this->actor.params]);
+    if (this->unk_194 != NULL) {
+        this->unk_194(this, globalCtx);
+    }
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+void EnOssan_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnOssan* this = THIS;
+
+    this->unk_1E4++;
+    this->actionFunc(this, globalCtx);
+}
+
+s32 func_80AC74F4(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnOssan* this = THIS;
+
+    if (limbIndex == 8) {
+        rot->x += this->unk_1EE;
+    }
+    return 0;
+}
+
+#ifdef NON_MATCHING
+// Small ordering
+void func_80AC7528(GlobalContext* globalCtx, EnOssan* this, f32 arg2, f32 arg3, f32 arg4, u8 arg5) {
+    f32 temp_f0;
+    f32 tmp2;
+
+    {
+        GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+        Gfx* dispRefs[4];
+
+        Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_en_oB1.c", 4192);
+        if (arg5) {
+            func_80094520(globalCtx->state.gfxCtx);
+
+            gDPSetPrimColor(gfxCtx->overlay.p++, 0, 0, this->unk_23C[0], this->unk_23C[1], this->unk_23C[2],
```

Comment:
unk_23C might be one of the structs in color.h

Reply:
Maybe, but struct accesses via . generate different code than array accesses via []. I've seen this over a bunch of files, array is required to match, and color.h doesn't have any arrays.